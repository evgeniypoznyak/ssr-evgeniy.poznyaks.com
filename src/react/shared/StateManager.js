import {createContext} from 'react';

export const sortSkills = skills => {
    const sortedSkills = skills.slice().sort(compareByOrder);
    return sortedSkills.reduce((acc, curr) => {
        if (curr['panel'] === 'left') {
            acc['left'].push(curr);
        }
        if (curr['panel'] === 'right') {
            acc['right'].push(curr);
        }
        return acc;
    }, {left: [], right: []});
};

export const compareByOrder = (a, b) => {
    if (a.order < b.order) {
        return -1;
    }
    if (a.order > b.order) {
        return 1;
    }
    return 0;
};

export const State = createContext(null);
