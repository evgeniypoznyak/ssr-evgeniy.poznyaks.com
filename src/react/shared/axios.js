import axios from 'axios';


axios.interceptors.request.use(
    config => {
        let token = '';
        // eslint-disable-next-line no-undef
        if (__isBrowser__) {
            if (localStorage) {
                token = localStorage.getItem('token');
            }
        }

        config.headers['x-auth-token'] = token;
        return config;
    },
    error => Promise.reject(error),
);

export default axios;
