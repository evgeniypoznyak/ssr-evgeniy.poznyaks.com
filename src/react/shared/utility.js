export const updateObject = (oldObject, updatedProperties) => {
    return {
        ...oldObject,
        ...updatedProperties,
    };
};

export const checkValidity = (value, rules) => {
    let isValid = true;
    if (!rules) {
        return true;
    }
    if (typeof (value) == 'undefined' || value == null) {
        return false;
    }

    if (rules.required) {
        isValid = ('' + value).trim() !== '' && isValid;
    }

    if (rules.minLength) {
        isValid = ('' + value).trim().length >= rules.minLength && isValid;
    }

    if (rules.maxLength) {
        const lenght = ('' + value).trim().length;
        isValid = lenght <= rules.maxLength && lenght > 0 && value !== '' && isValid;
    }

    if (rules.isEmail) {
        // eslint-disable-next-line max-len
        const pattern = /[a-z0-9!#$%&'*+/=?^_`{|}~-]+(?:\.[a-z0-9!#$%&'*+/=?^_`{|}~-]+)*@(?:[a-z0-9](?:[a-z0-9-]*[a-z0-9])?\.)+[a-z0-9](?:[a-z0-9-]*[a-z0-9])?/;
        isValid = pattern.test(value) && isValid;
    }

    if (rules.isNumeric) {
        const pattern = /^\d+$/;
        isValid = pattern.test(value) && isValid;
    }

    return isValid;
};

export const browserLanguage = function browserLocale() {
    if (navigator.languages && navigator.languages.length) {
        return navigator.languages[0];
    } else if (navigator.userLanguage) {
        return navigator.userLanguage;
    }
    return navigator.language;
};

export const stripHtmlTags = element => element.replace(/<\/?[^>]+(>|$)/g, '');
