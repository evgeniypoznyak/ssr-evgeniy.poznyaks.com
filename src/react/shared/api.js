import axios from './axios';
import {skills} from './initialData/skills';

const url = process.env.API_GATEWAY ||
    'https://api.evgeniy.poznyaks.com'
    // 'http://localhost:2222'
;
const apiPublic = process.env.API_GATEWAY_PUBLIC ||
    'https://api.evgeniy.poznyaks.com'
    // 'http://localhost:2222'
;

const selfUrl = process.env.SELF_URL ||
    ''
    // 'http://localhost:3001'
;

export async function fetchSkills(path = 'all') {
    const biography = await getBiography();
    try {
        console.log('[GET] processing url: ', `${url}/api/skills/`);
        const result = await axios.get(`${url}/api/skills/`);
        return {
            skills: result.data.skills,
            biography,
        };
    } catch (e) {
        console.log('Error: ', e.message);
        console.log('Something wrong happened with API Gateway! Returning default response.');
        console.log('returning regular skills...');
        return {skills, biography};
    }
}

export async function sendEmail(data) {
    try {
        const url = `${apiPublic}/api/email`;
        console.log('[POST] processing url: ', `${url}`);
        const result = await axios.post(url, data);
        return result.data;
    } catch (e) {
        const error = {
            status: e.response.data.status,
            message: e.response.data.message,
            errorMessage: e.message,
        };
        console.log('Error: ', e.message);
        console.log('Something wrong happened with API Gateway! Returning default response.');
        return error;
    }
}

export async function fetchBiography(path) {
    console.log(path);
    const url = '/biography';
    try {
        console.log('[GET] processing url: ', `${url}/`);
        const result = await axios.get(`${url}/`);
        return result.data;
    } catch (e) {
        console.log('Error: ', e.message);
        console.log('Something wrong happened with API Gateway! Returning default response.');
        console.log('returning regular skills...');
        return {skills};
    }
}

export async function signInIntoApiGateway(data) {
    try {
        console.log('[POST] processing url: ', `${apiPublic}/api/auth`);
        const result = await axios.post(`${apiPublic}/api/auth`, {...data});
        return result.data;
    } catch (e) {
        console.log('Error: ', e.message);
        console.log('Something wrong happened with API Gateway! Returning default response.');
        return {token: ''};
    }
}

export async function verifyToken() {
    try {
        // eslint-disable-next-line no-undef
        if (__isBrowser__ && localStorage.getItem('token') && localStorage.getItem('token').length > 0) {
            return 'OK';
        }
    } catch (e) {
        console.log('Error: ', e.message);
        console.log('Something wrong happened with API Gateway! Returning default response.');
        return {token: ''};
    }
}

export async function verifyTokenApi() {
    try {
        const url = `${apiPublic}/api/auth/verify`;
        console.log('[POST] processing url: ', `${url}`);
        const result = await axios.post(url);
        return result.data;
    } catch (e) {
        console.log('Error: ', e.message);
        console.log('Something wrong happened with API Gateway! Returning default response.');
        return {token: ''};
    }
}

export async function patchSkill(skill) {
    try {
        const url = `${apiPublic}/api/skills`;
        console.log('[PATCH] processing url: ', `${url}`);
        const result = await axios.patch(url, {skill});
        return result.data;
    } catch (e) {
        console.log('Error: ', e.message);
        console.log('Something wrong happened with API Gateway! Returning default response.');
        return 'Skill not been patched';
    }
}

export async function deleteSkill(skillId) {
    try {
        const url = `${apiPublic}/api/skills/` + skillId;
        console.log('[DELETE] processing url: ', `${url}`);
        const result = await axios.delete(url);
        return result.data;
    } catch (e) {
        console.log('Error: ', e.message);
        console.log('Something wrong happened with API Gateway! Returning default response.');
        return 'Skill not been deleted';
    }
}

export async function uploadResume(resume) {
    try {
        const verification = await verifyTokenApi();
        if (verification === 'OK') {
            const url = '/resume';
            console.log('[POST] processing url: ', `${url}`);
            return await axios.post(url, resume);
        }
    } catch (e) {
        console.log('Error: ', e.message);
        console.log('Something wrong happened with this server! Returning default response.');
        return 'Resume not been uploaded';
    }
}


export async function putSkills(skills) {
    try {
        const verification = await verifyTokenApi();
        if (verification === 'OK') {
            const url = `${apiPublic}/api/skills`;
            console.log('[PATCH] processing url: ', `${url}`);
            const result = await axios.put(url, {skills});
            return result.data;
        }
    } catch (e) {
        console.log('Error: ', e.message);
        console.log('Something wrong happened with API Gateway! Returning default response.');
        return 'Skill not been patched';
    }
}


export async function uploadBiography(biography) {
    try {
        const verification = await verifyTokenApi();
        if (verification === 'OK') {
            const url = '/biography';
            console.log('[POST] processing url: ', `${url}`);
            return await axios.post(url, {biography});
        }
    } catch (e) {
        console.log('Error: ', e.message);
        console.log('Something wrong happened with this server! Returning default response.');
        return 'Resume not been uploaded';
    }
}

export async function getBiography() {
    try {
        const url = `${selfUrl}/biography`;
        console.log('[GET] processing url: ', `${url}`);
        const result = await axios.get(url);
        return result.data.biography;
    } catch (e) {
        console.log('Error: ', e.message);
        console.log('Something wrong happened with this server! Returning default response.');
        return 'Biography not been uploaded';
    }
}


export const getSkillsByPanes = skills => {
    const sortedSkills = skills.slice().sort(sortByOrder);
    return sortedSkills.reduce((acc, curr) => {
        if (curr['panel'] === 'left') {
            acc['left'].push(curr);
        }
        if (curr['panel'] === 'right') {
            acc['right'].push(curr);
        }
        return acc;
    }, {left: [], right: []});
};

export const sortByOrder = (a, b) => {
    if (a.order * 1 < b.order * 1) {
        return -1;
    }
    if (a.order * 1 > b.order * 1) {
        return 1;
    }
    return 0;
};
