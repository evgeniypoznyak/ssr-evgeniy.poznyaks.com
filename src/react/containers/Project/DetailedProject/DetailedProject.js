import React, {Fragment} from 'react';
import {makeStyles} from '@material-ui/core/styles';
import Paper from '@material-ui/core/Paper';
import {getRandomString} from '../../../shared/utility';
import {Grid} from '@material-ui/core';


const useStyles = makeStyles(theme => ({
    root: {
        padding: theme.spacing(3, 2),
    },
    paper: {
        'padding': theme.spacing(3, 2),
        'fontSize': '2rem',
        '@media (max-width: 960px)': {
            fontSize: '1.3rem',
        },
        '@media (max-width: 600px)': {
            fontSize: '1rem',
        },
    },
    imageWrapper: {
        width: '100%',
        textAlign: 'center',
    },
    image: {
        maxWidth: '100%',
    },
    descriptionElement: {
        marginBottom: 20,
        padding: 10,
    },
    divider: {
        marginTop: 20,
        marginBottom: 20,
    },
}));

export default function DetailedProject(props) {
    const classes = useStyles();

    return (
        <Fragment>
            {props.description.map(d => {
                return (
                    <Fragment key={getRandomString(99)}>
                        <Paper className={classes.descriptionElement}>
                            <Grid
                                container
                                direction="row"
                                justify="center"
                                spacing={1}
                                alignItems="center">
                                <Grid item xs={12} md={6}>
                                    {d.picture ? <Fragment>
                                        <div className={classes.imageWrapper}>
                                            <img alt="detailed" src={d.picture} className={classes.image}/>
                                        </div>
                                    </Fragment> : null}
                                </Grid>
                                <Grid item xs={12} md={6}>
                                    <Paper className={classes.paper}>{d.description}</Paper>
                                </Grid>
                            </Grid>
                        </Paper>
                    </Fragment>
                );
            })}
        </Fragment>
    );
}
