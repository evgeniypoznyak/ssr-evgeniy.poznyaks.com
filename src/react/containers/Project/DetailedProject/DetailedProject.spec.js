import React from 'react';
import {shallow} from 'enzyme';
import {skills} from '../../../shared/api';
import {Divider, Paper} from '@material-ui/core';
import DetailedProject from './DetailedProject';

let wrapper;

const longDescription = skills.slice()[0].projects[0].longDescription;

beforeAll(() => {
    wrapper = shallow(<DetailedProject description={longDescription}/>);
});

describe('<DetailedProject/>', () => {
    it('renders correctly', () => {
        expect(wrapper.children.length).toBe(1);
        expect(wrapper.getElement()).toMatchSnapshot();
    });

    it('renders Paper', () => {
        expect(wrapper.find(Paper)).toHaveLength(2);
    });

    it('renders Divider', () => {
        expect(wrapper.find(Divider)).toHaveLength(2);
    });
});
