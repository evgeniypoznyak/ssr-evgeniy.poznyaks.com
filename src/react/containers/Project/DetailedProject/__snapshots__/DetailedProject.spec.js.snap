// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<DetailedProject/> renders correctly 1`] = `
<React.Fragment>
  <React.Fragment>
    <ForwardRef(WithStyles)
      className="makeStyles-descriptionElement-4"
    >
      <React.Fragment>
        <div
          className="makeStyles-imageWrapper-2"
        >
          <img
            alt="detailed"
            className="makeStyles-image-3"
            src="https://hygger.io/blog/wp-content/uploads/2018/05/Scrum-3.jpg"
          />
        </div>
      </React.Fragment>
      <ForwardRef(WithStyles)
        className="makeStyles-divider-5"
        component="hr"
        variant="middle"
      />
      <div>
        In very beginning I created PHP website from scratch. I didn't choice any library or framework, it was in pure php. In very beginning I created PHP website from scratch. I didn't choice any library or framework, it was in pure php.
      </div>
    </ForwardRef(WithStyles)>
  </React.Fragment>
  <React.Fragment>
    <ForwardRef(WithStyles)
      className="makeStyles-descriptionElement-4"
    >
      <React.Fragment>
        <div
          className="makeStyles-imageWrapper-2"
        >
          <img
            alt="detailed"
            className="makeStyles-image-3"
            src="https://hygger.io/blog/wp-content/uploads/2018/05/Scrum-3.jpg"
          />
        </div>
      </React.Fragment>
      <ForwardRef(WithStyles)
        className="makeStyles-divider-5"
        component="hr"
        variant="middle"
      />
      <div>
        In very beginning I created PHP website from scratch. I didn't choice any library or framework, it was in pure php. In very beginning I created PHP website from scratch. I didn't choice any library or framework, it was in pure php.
      </div>
    </ForwardRef(WithStyles)>
  </React.Fragment>
</React.Fragment>
`;
